crypto/evp/p_lib.c

```diff
@@ -1386,17 +1386,21 @@ static int evp_pkey_reset_unlocked(EVP_PKEY *pk)
     if (pk == NULL)
         return 0;
 
-    memset(pk, 0, sizeof(*pk));
+    if (pk->lock != NULL) {
+      const size_t offset = (unsigned char *)&pk->lock - (unsigned char *)pk;
+
+      memset(pk, 0, offset);
+      memset((unsigned char *)pk + offset + sizeof(pk->lock),
+             0,
+             sizeof(*pk) - offset - sizeof(pk->lock));
+    }
+    /* EVP_PKEY_new will have zalloced the memory so no need to call memset */
```

Comment:
Nit: Please amend the comment to something like: `EVP_PKEY_new uses zalloc so no need to call memset if pk->lock is NULL.`

Reply:
Update now, thanks